### PR TITLE
Popup: fix findDOMNode call inside componentWillUnmount hook

### LINKED ISSUE #114: Popup:  exception findDOMNode was called on an unmounted component
Срабатывает при переходе на другую страницу(react-router), если попап был открыт хотя бы один раз. 

https://github.com/narqo/react-islands/blob/master/src/Popup/index.js#L109

### PR DESCRIPTION
**Pull request by narqo**
fix #114

/cc @fresk-nc

add lodash/throttle to isAnchorVisible

### CODE REVIEW

File: src/Popup/index.js [hunk 1]
```diff
@@ -150,15 +153,19 @@ class Popup extends Component {
...
+        // TODO(@narqo): don't call DOMNode measurements in case nothing has changed
+        const layout = this.calcBestLayoutParams();
+
+        if (this.props.onLayout) {
+            this.props.onLayout({ layout }, this.props);
+        }
 
-            const { direction, left, top } = layout;
+        const { direction, left, top } = layout;
+        if (this.state.direction !== direction || this.state.left !== left || this.state.top !== top) {
```
Comment (narqo) [inline src/Popup/index.js:168]: sometimes reactive programming sucks :goberserk:

**Comment by fresk-nc** [comment]
👍

File: src/Popup/index.js (hunk 1 above)
Comment (vodkabears) [reply src/Popup/index.js:168]: Try shallow compare: https://facebook.github.io/react/docs/shallow-compare.html

Comment (narqo) [reply src/Popup/index.js:168]: The problem is not how the comparison is made, but in the way how this peace is implemented. We have to call `reposition` in `componentDidMount`, as it must have an access to the re-rendered DOM, but, at the same time, we have to call `setState` here, as we must update CSS-classes and styles (and I don't want to do it directly in DOM). This overall leads to inefficient code and lots of hacks :(

**Pull request merged**